Source/Core/VideoCommon/PerformanceTracker.h

```diff
@@ -89,16 +78,15 @@ class PerformanceTracker
   // Queue + Running Total used to calculate average dt
   DT m_dt_total = DT::zero();
   std::array<DT, MAX_DT_QUEUE_SIZE> m_dt_queue;
-  std::size_t m_dt_queue_begin = 0;
-  std::size_t m_dt_queue_end = 0;
+  u64 m_dt_queue_begin = 0, m_dt_queue_end = 0;
 
   // Average rate/time throughout the window
   DT m_dt_avg = DT::zero();  // Uses Moving Average
   double m_hz_avg = 0.0;     // Uses Moving Average + Euler Average
 
   // Used to initialize this on demand instead of on every Count()
-  mutable std::optional<DT> m_dt_std = std::nullopt;
+  mutable std::optional<DT> m_dt_std;
 
   // Used to enable thread safety with the performance tracker
-  mutable std::mutex m_mutex;
+  mutable std::shared_mutex m_mutex;
```

Comment:
This mutex change should definitely be a separate commit.

Reply:
I will address the comments tomorrow or another time because it is currently 1am, but the TLDR is that these changes come directly from a review of a previous commit. Because there is a chain of PR's, its easiest to fix the issues delroth mentioned in the previous PR here.